cadence/contracts/bridge/FlowEVMBridgeHandlers.cdc

```diff
@@ -0,0 +1,243 @@
+import "Burner"
+import "FungibleToken"
+import "NonFungibleToken"
+
+import "EVM"
+
+import "EVMUtils"
+import "FlowEVMBridgeHandlerInterfaces"
+import "FlowEVMBridgeConfig"
+import "FlowEVMBridgeUtils"
+
+/// FlowEVMBridgeHandlers
+///
+/// This contract is responsible for defining and configuring bridge handlers for special cased assets.
+///
+access(all) contract FlowEVMBridgeHandlers {
+
+    /**********************
+        Contract Fields
+    ***********************/
+
+    /// The storage path for the HandlerConfigurator resource
+    access(all) let ConfiguratorStoragePath: StoragePath
+
+    /****************
+        Constructs
+    *****************/
+
+    /// Handler for bridging Cadence native fungible tokens to EVM. In the event a Cadence project migrates native
+    /// support to EVM, this Hander can be configured to facilitate bridging the Cadence tokens to EVM. This Handler
+    /// then effectively allows the bridge to treat such tokens as bridge-defined on the Cadence side and EVM-native on
+    /// the EVM side minting/burning in Cadence and escrowing in EVM.
+    /// In order for this to occur, neither the Cadence token nor the EVM contract can be onboarded to the bridge - in
+    /// essence, neither side of the asset can be onboarded to the bridge.
+    /// The Handler must be configured in the bridge via the HandlerConfigurator. Once added, the bridge will filter
+    /// requests to bridge the token Vault to EVM through this Handler which cannot be enabled until a target EVM 
+    /// address is set. Once the corresponding EVM contract address is known, it can be set and the Handler. It's also
+    /// suggested that the Handler only be enabled once sufficient liquidity has been arranged in bridge escrow on the
+    /// EVM side.
+    ///
+    access(all) resource CadenceNativeTokenHandler : FlowEVMBridgeHandlerInterfaces.TokenHandler {
+        /// Flag determining if request handling is enabled
+        access(self) var enabled: Bool
+        /// The Cadence Type this handler fulfills requests for
+        access(self) var targetType: Type
+        /// The EVM contract address this handler fulfills requests for. This field is optional in the event the EVM
+        /// contract address is not yet known but the Cadence type must still be filtered via Handler to prevent the
+        /// type from being onboarded otherwise.
+        access(self) var targetEVMAddress: EVM.EVMAddress?
+        /// The Minter enabling minting of Cadence tokens on fulfillment from EVM
+        access(self) let minter: @{FlowEVMBridgeHandlerInterfaces.TokenMinter}
+
+        init(targetType: Type, targetEVMAddress: EVM.EVMAddress?, minter: @{FlowEVMBridgeHandlerInterfaces.TokenMinter}) {
```

Comment:
I think that it would be better if the minter resource could just be sent via a public function in a public capability, like `addMinterCapability()` and it would check to see that the type and such are correct so that people can't sent junk minters to it. That way, the minter can be sent trustlessly, which I think will be important for some of these projects

Reply:
That's a good idea. So do something like add an `expectedMinterType` on init instead of the minter itself and confirm that the minter added in `addMinterCapability` is the expected type?

Alternatively, we could entitle the `addMinterCapability` and publish a private Capability to the minter provider so they can privately insert the minter. It closes the loop a bit more but does require a bit more coordination.

Regardless, I like that either option gives us the ability to reserve a handler for a given type without first needing a minter resource.

Reply:
I like the option that requires the least coordination, so I think just it being a public function that checks the minter type is sufficient

Reply:
Updated to address in [3f5d19e](https://github.com/onflow/flow-evm-bridge/pull/46/commits/3f5d19e4fce4d38e5d13c481ca95e4dc211b9b4e)